src/parser.js

```diff
@@ -822,34 +822,49 @@ export default class Parser {
 
     examples = _.omit(examples, 'schema');
 
-    _.forEach(examples, (responseBody, contentType) => {
-      const transaction = this.createTransaction(transition, method, schemes);
+    const consumes = methodValue.consumes || this.swagger.consumes || [];
+    const produces = methodValue.produces || this.swagger.produces || [];
+
+    let jsonConsumes = _.filter(consumes, isJsonContentType);
+    let jsonProduces = _.filter(produces, isJsonContentType);
+
+    if (!jsonProduces.length) {
+      jsonProduces = jsonProduces.concat(null);
+    }
 
-      this.handleSwaggerExampleRequest(transaction, methodValue, transitionParams, resourceParams);
-      this.handleSwaggerExampleResponse(transaction, methodValue, responseValue,
-                                        statusCode, responseBody, contentType);
+    if (!jsonConsumes.length) {
+      jsonConsumes = jsonConsumes.concat(null);
+    }
+
+    _.forEach(examples, (responseBody, contentType) => {
```

Comment:
This will only generate request/response pairs if they are JSON or empty. This will cause content types such as `application/x-www-form-urlencoded`, `multipart/form-data` to not generate a request/response.